plugin/kubernetes/pvc.go

```diff
@@ -135,13 +135,15 @@ func (v *pvc) inContainer(ctx context.Context, fn containerCb) (interface{}, err
 		if err != nil {
 			return nil, err
 		}
+		cleanup = func() {
+			// Use a background context to ensure deletion happens even if context was cancelled.
+			activity.Record(ctx, "Deleted temporary pod %v: %v", tempPod, tempPod.delete(context.Background()))
```

Comment:
Why background? Why not a timeout context?

Reply:
Hmm. I guess. I'd hope the Delete implementation would have a somewhat bounded runtime, especially if it also supports context cancellation.

Reply:
Same comment would also apply to https://github.com/puppetlabs/wash/blob/master/plugin/docker/volume.go#L122-L129.

Reply:
Ah yeah, I missed that one. I guess the Q. here is "Does this code introduce a theoretically-possible edge-case of a dangling go-routine?" If so, may as well fix it since adding a timeout context is easy. If not, then no need.

Either way not a blocker. Rest of the PR looks good to me.

Reply:
The timeout context is no guarantee either, it'll only halt the goroutine if the code that's blocking also checks the context for cancelation.